tests/devices/qubit_mixed/test_qubit_mixed_simulate.py

```diff
@@ -273,8 +274,225 @@ def test_broadcasting_with_extra_measurement_wires(self, mocker):
         assert spy.call_args_list[0].args == (qs, {0: 0, 2: 1})
 
 
-def test_finite_shot_not_implemented():
-    op = qml.RX(np.pi, [0])
-    qs = qml.tape.QuantumScript([op], [qml.expval(qml.Z(0))], shots=1000)
-    with pytest.raises(NotImplementedError):
-        simulate(qs)
+@flaky(max_runs=5, min_passes=1)
+class TestSampleMeasurements:
+    """Tests circuits with sample-based measurements"""
+
+    @staticmethod
+    def expval_of_RY_circ(x):
+        """Find the expval of PauliZ on simple RY circuit"""
+        return np.cos(x)
+
+    @staticmethod
+    def sample_sum_of_RY_circ(x):
+        """Find the expval of computational basis bitstring value for both wires on simple RY circuit"""
+        return [np.sin(x / 2) ** 2, 0]
+
+    @staticmethod
+    def expval_of_2_qubit_circ(x):
+        """Gets the expval of PauliZ on wire=0 on the 2-qubit circuit used"""
+        return np.cos(x)
+
+    @staticmethod
+    def probs_of_2_qubit_circ(x, y):
+        """Possible measurement values and probabilities for the 2-qubit circuit used"""
+        probs = (
+            np.array(
+                [
+                    np.cos(x / 2) * np.cos(y / 2),
+                    np.cos(x / 2) * np.sin(y / 2),
+                    np.sin(x / 2) * np.sin(y / 2),
+                    np.sin(x / 2) * np.cos(y / 2),
+                ]
+            )
+            ** 2
+        )
+        return ["00", "01", "10", "11"], probs
+
+    @pytest.mark.parametrize("interface", ml_interfaces)
+    @pytest.mark.parametrize("x", [0.732, 0.488])
+    def test_single_expval(self, x, interface, seed):
+        """Test a simple circuit with a single expval measurement"""
+        qs = qml.tape.QuantumScript(
+            [qml.RY(x, wires=0)],
+            [qml.expval(qml.PauliZ(0))],
+            shots=10000,
+        )
+        result = simulate(qs, rng=seed, interface=interface)
+        assert isinstance(result, np.float64)
+        assert result.shape == ()
+        assert np.allclose(result, self.expval_of_RY_circ(x), atol=0.05)
```

Comment:
Personally, I would want to restrict the tests for numerical correctness (especially when shots are present) to the unit tests of the individual components that are actually generating the numerical results. In this case, if `measure_with_samples` was properly tested for numerical correctness, we shouldn't need this many numerical tests here. I would say simply test different combinations of scenarios with/without shot vectors and with/without parameter broadcasting and single/multiple measurements, etc., to see that the results have the correct shapes.

Reply:
Nice. I'll make changes soon

Reply:
Removed the numeric checks for finite-shot simulate tests